src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -1681,6 +1681,19 @@ def _openssh_public_key_bytes(self, key):
             )
 
 
+# If the backend has support for Scrypt, dynamically register the backend and
+# implement the interface.
+def _derive_scrypt(self, key_material, salt, length, n, r, p):
+    buf = self._ffi.new("unsigned char[]", length)
+    self._lib.EVP_PBE_scrypt(key_material, len(key_material), salt,
+                             len(salt), n, r, p, 0, buf, length)
+    return self._ffi.buffer(buf)[:]
+
+if binding.Binding().lib.Cryptography_HAS_SCRYPT == 1:
+    ScryptBackend.register(Backend)
```

Comment:
I'd write a `@utils.register_interface_if(lib.Cryptography_HAS_SCRYPT == 1, ScryptBackend)` function, which I think makes this much cleaner.